presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HistoryTable.java

```diff
@@ -0,0 +1,97 @@
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.InMemoryRecordSet;
+import io.prestosql.spi.connector.RecordCursor;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.spi.type.Type;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static java.util.Objects.requireNonNull;
+
+public class HistoryTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = new ImmutableList.Builder<ColumnMetadata>().add(
+            new ColumnMetadata("made_current_at", TIMESTAMP),
+            new ColumnMetadata("snapshot_id", BIGINT),
+            new ColumnMetadata("parent_id", BIGINT),
+            new ColumnMetadata("is_current_ancestor", BOOLEAN)).build();
+
+    public HistoryTable(SchemaTableName tableName, Table icebergTable)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return new ConnectorTableMetadata(tableName, COLUMNS);
+    }
+
+    @Override
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        ImmutableList.Builder<List<Object>> records = ImmutableList.builder();
+        Set<Long> ancestorIds = ImmutableSet.copyOf(SnapshotUtil.currentAncestors(icebergTable));
+        for (HistoryEntry historyEntry : icebergTable.history()) {
+            long snapshotId = historyEntry.snapshotId();
+            Snapshot snapshot = icebergTable.snapshot(snapshotId);
+
+            List<Object> record = new ArrayList<>();
+            record.add(historyEntry.timestampMillis());
+            record.add(snapshotId);
+            record.add(snapshot != null ? snapshot.parentId() : null);
+            record.add(ancestorIds.contains(snapshotId));
+
+            records.add(record);
+        }
+
+        List<Type> types = COLUMNS.stream()
```

Comment:
It is not necessary to compute `types` every time. Change it to `static final`.

Reply:
I think this method will only be called once during execution?